dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/deduplication/DeduplicationController.java

```diff
@@ -159,18 +151,18 @@ public PotentialDuplicate postPotentialDuplicate(
 
     @PutMapping( value = "/{id}" )
     @ResponseStatus( value = HttpStatus.OK )
-    public void updatePotentialDuplicate( @PathVariable String id, @RequestParam( value = "status" ) String status )
+    public void updatePotentialDuplicate( @PathVariable String id,
+        @RequestParam( value = "status" ) DeduplicationStatus status )
         throws NotFoundException,
         BadRequestException
     {
         checkDeduplicationStatusRequestParam( status );
```

Comment:
This check will never throw an Exception. If an invalid value is passed for the DeduplicationStatus then Spring will throw a `MethodArgumentTypeMismatchException` that will be intercepted in `CrudControllerAdvice` and translated to a `BadRequestException`